File: Java/dijkstra/llama-3.3-70b-versatile/Llama_3_3_70b_versatile.java
import java.util.*;

public class Llama_3_3_70b_versatile {
    private static class Node implements Comparable<Node> {
        int id;
        int distance;

        public Node(int id, int distance) {
            this.id = id;
            this.distance = distance;
        }

        @Override
        public int compareTo(Node other) {
            return Integer.compare(this.distance, other.distance);
        }
    }

    public static void dijkstra(int[][] graph, int startNode) {
        int numNodes = graph.length;
        int[] distances = new int[numNodes];
        int[] previousNodes = new int[numNodes];
        boolean[] visited = new boolean[numNodes];

        // Initialize distances and previous nodes
        for (int i = 0; i < numNodes; i++) {
            distances[i] = Integer.MAX_VALUE;
            previousNodes[i] = -1;
        }
        distances[startNode] = 0;

        // Create a priority queue to hold nodes to visit
        PriorityQueue<Node> queue = new PriorityQueue<>();
        queue.add(new Node(startNode, 0));

        while (!queue.isEmpty()) {
            Node currentNode = queue.poll();
            int currentNodeId = currentNode.id;

            // If we've already visited this node, skip it
            if (visited[currentNodeId]) {
                continue;
            }

            // Mark the node as visited
            visited[currentNodeId] = true;

            // Update distances for neighboring nodes
            for (int i = 0; i < numNodes; i++) {
                int edgeWeight = graph[currentNodeId][i];
                if (edgeWeight > 0 && !visited[i]) {
                    int newDistance = distances[currentNodeId] + edgeWeight;
                    if (newDistance < distances[i]) {
                        distances[i] = newDistance;
                        previousNodes[i] = currentNodeId;
                        queue.add(new Node(i, newDistance));
                    }
                }
            }
        }

        // Print the shortest distances and paths
        System.out.println("Shortest Distances:");
        for (int i = 0; i < numNodes; i++) {
            System.out.println("Node " + i + ": " + distances[i]);
        }

        System.out.println("Shortest Paths:");
        for (int i = 0; i < numNodes; i++) {
            if (i != startNode) {
                System.out.print("Path from node " + startNode + " to node " + i + ": ");
                printPath(previousNodes, startNode, i);
                System.out.println();
            }
        }
    }

    private static void printPath(int[] previousNodes, int startNode, int endNode) {
        if (endNode == startNode) {
            System.out.print(startNode);
        } else {
            printPath(previousNodes, startNode, previousNodes[endNode]);
            System.out.print(" -> " + endNode);
        }
    }

    public static void main(String[] args) {
        int numNodes = 1000;

        // Gerador de números aleatórios
        Random random = new Random();

        // Matriz de adjacência para o grafo
        int[][] graph = new int[numNodes][numNodes];

        // Criando arestas aleatórias
        for (int i = 0; i < numNodes; i++) {
            for (int j = i + 1; j < numNodes; j++) {
                if (random.nextDouble() < 0.05) { // 5% de chance de criar uma aresta
                    int weight = random.nextInt(10) + 1;  // Peso aleatório entre 1 e 10
                    graph[i][j] = weight;
                    graph[j][i] = weight;  // Grafo não direcionado
                }
            }
        }

        // Nó de partida aleatório
        int startNode = random.nextInt(numNodes);

        // Calculando as distâncias usando o algoritmo de Dijkstra
        dijkstra(graph, startNode);
    }
}

File: Python/dijkstra/llama-3.3-70b-versatile/llama-3.3-70b-versatile.py
import sys
import heapq
import random
def dijkstra(graph, start_node):
    """
    Dijkstra's algorithm implementation.

    Args:
    - graph (dict): A dictionary representing the graph, where each key is a node and its corresponding value is another dictionary.
                    The inner dictionary's keys are the neighboring nodes, and its values are the edge weights.
    - start_node (str): The node to start the search from.

    Returns:
    - distances (dict): A dictionary containing the shortest distances from the start node to all other nodes.
    - previous_nodes (dict): A dictionary containing the previous node in the shortest path from the start node to each node.
    """

    # Initialize distances and previous nodes dictionaries
    distances = {node: sys.maxsize for node in graph}
    distances[start_node] = 0
    previous_nodes = {node: None for node in graph}

    # Create a priority queue
    priority_queue = [(0, start_node)]

    while priority_queue:
        # Extract the node with the minimum distance from the priority queue
        current_distance, current_node = heapq.heappop(priority_queue)

        # If the current distance is greater than the already known distance, skip this node
        if current_distance > distances[current_node]:
            continue

        # Iterate over the neighbors of the current node
        for neighbor, weight in graph[current_node].items():
            distance = current_distance + weight

            # If a shorter path to the neighbor is found, update the distance and previous node
            if distance < distances[neighbor]:
                distances[neighbor] = distance
                previous_nodes[neighbor] = current_node
                heapq.heappush(priority_queue, (distance, neighbor))

    return distances, previous_nodes


def reconstruct_path(previous_nodes, start_node, end_node):
    """
    Reconstruct the shortest path from the start node to the end node.

    Args:
    - previous_nodes (dict): A dictionary containing the previous node in the shortest path from the start node to each node.
    - start_node (str): The node to start the search from.
    - end_node (str): The node to end the search at.

    Returns:
    - path (list): A list of nodes representing the shortest path from the start node to the end node.
    """

    path = []
    current_node = end_node

    while current_node is not None:
        path.append(current_node)
        current_node = previous_nodes[current_node]

    # Reverse the path to get it in the correct order
    path.reverse()

    return path


def generate_random_graph(num_nodes):
    graph = {str(i): {} for i in range(num_nodes)}
    for i in range(num_nodes):
        for j in range(i + 1, num_nodes):
            weight = random.randint(1, 100)
            graph[str(i)][str(j)] = weight
            graph[str(j)][str(i)] = weight
    return graph

def main():
    num_nodes = 1000
    graph = generate_random_graph(num_nodes)
    start_node = '0'
    dijkstra(graph, start_node)

if __name__ == "__main__":
    main()